pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+	objectsToPropagate := make([]*unstructured.Unstructured, 0)
+	// add cluster-scoped objects to the 'objectsToPropagate' slice
+	for _, clusterScopedObject := range placementDecision.Spec.Workload.ClusterScope {
+		if clusterScopedObject.ObjectNames == nil {
+			continue // no objects from this gvr, skip
+		}
+		gvr := schema.GroupVersionResource{Group: clusterScopedObject.Group, Version: clusterScopedObject.Version, Resource: clusterScopedObject.Resource}
+		for _, objectName := range clusterScopedObject.ObjectNames {
+			object, err := c.wdsDynamicClient.Resource(gvr).Get(ctx, objectName, metav1.GetOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get required cluster-scoped object '%s' with gvr %s from WDS - %w", objectName, gvr, err)
+			}
+			objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+		}
+	}
+	// add namespace-scoped objects to the 'objectsToPropagate' slice
+	for _, namespaceScopedObject := range placementDecision.Spec.Workload.NamespaceScope {
+		gvr := schema.GroupVersionResource{Group: namespaceScopedObject.Group, Version: namespaceScopedObject.Version, Resource: namespaceScopedObject.Resource}
+		for _, objectsByNamespace := range namespaceScopedObject.ObjectsByNamespace {
+			if objectsByNamespace.Names == nil {
+				continue // no objects from this namespace, skip
+			}
+			for _, objectName := range objectsByNamespace.Names {
+				object, err := c.wdsDynamicClient.Resource(gvr).Namespace(objectsByNamespace.Namespace).Get(ctx, objectName, metav1.GetOptions{})
+				if err != nil {
+					return nil, fmt.Errorf("failed to get required namespace-scoped object '%s' with gvr '%s' from WDS - %w", objectName, gvr, err)
+				}
+				objectsToPropagate = append(objectsToPropagate, cleanObject(object))
+			}
+		}
+	}
+
+	return objectsToPropagate, nil
+}
+
+func (c *genericTransportController) createOrUpdateWrappedObject(ctx context.Context, namespace string, wrappedObject *unstructured.Unstructured) error {
+	existingWrappedObject, err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Get(ctx, wrappedObject.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if !strings.HasSuffix(err.Error(), notFoundErrorSuffix) { // if object is not there, we need to create it. otherwise report an error.
+			return fmt.Errorf("failed to create wrapped object '%s' in destination WEC with namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+		}
+		// object not found when using get, create it
+		_, err = c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Create(ctx, wrappedObject, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+		}
+		return nil
+	}
+	// // if we reached here object already exists, try update object
+	wrappedObject.SetResourceVersion(existingWrappedObject.GetResourceVersion())
```

Comment:
What goes wrong if this line is omitted?

Reply:
if we don't set the resource version update fails with the following error:
```
metadata.resourceVersion: Invalid value: "": must be specified for an update
```

Reply:
Thanks.